src/Directions.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react'
+
+import { MapContext } from './Map'
+
+import { createCoordinate, NumberTuple } from './utils'
+
+type annotationOptions = {
+  origin?: mapkit.MarkerAnnotationConstructorOptions
+  destination?: mapkit.MarkerAnnotationConstructorOptions
+}
+
+type DirectionsProps = {
+  origin: NumberTuple
+  destination: NumberTuple
+  transportType?: 'Automobile' | 'Walking'
+  style?: mapkit.StyleConstructorOptions
+  options?: mapkit.DirectionsConstructorOptions
+  showAnnotations?: boolean
+  annotationOptions?: annotationOptions
+  routeDetails?: (arg0: {
+    steps: mapkit.RouteStep[]
+    distance: number
+    expectedTravelTime: number
+  }) => void
+}
+
+export const Directions: React.FC<DirectionsProps> = ({
+  origin,
+  destination,
+  transportType = 'Automobile',
```

Comment:
should we update this to take `mapkit.Directions.Transport.Automobile` instead of a hardcoded string?

Reply:
Due to the reason stated above regarding the type, this could be done within the component by introducing an if statement, checking if `request.transportType` has been set, else default to `mapkit.Directions.Transport.Automobile`